File: src/mffs/DelayedEvent.java
package mffs;

public abstract class DelayedEvent
{
	public int ticks = 0;
	protected IDelayedEventHandler handler;

	public DelayedEvent(IDelayedEventHandler handler, int ticks)
	{
		this.handler = handler;
		this.ticks = ticks;
	}

	protected abstract void onEvent();

	public void update()
	{
		this.ticks--;

		if (this.ticks <= 0)
		{
			this.onEvent();
		}
	}

	/**
	 * The higher the number, the higher the priority.
	 * 
	 * @return
	 */
	public int getPriority()
	{
		return 0;
	}
}

File: src/main/scala/mffs/field/mobilize/event/DelayedEvent.scala
package mffs.field.mobilize.event

class DelayedEvent(val handler: IDelayedEventHandler, var ticks: Int, val evtMethod: (() => Unit) = null)
{
  protected def onEvent = evtMethod.apply()

  def update()
  {
    if (ticks == 0)
    {
      onEvent
    }

    ticks -= 1
  }

  /**
   * The higher the number, the higher the priority.
   * @return
   */
  def priority: Int = 0
}